server/src/services/search.service.ts

```diff
@@ -144,6 +156,97 @@ export class SearchService {
     }
   }
 
+  async handleQueueSearchDuplicates({ force }: IBaseJob): Promise<JobStatus> {
+    const { machineLearning } = await this.configCore.getConfig();
+    if (!machineLearning.enabled || !machineLearning.clip.enabled) {
```

Comment:
```suggestion
    if (await this.configCore.hasFeature(FeatureFlag.SMART_SEARCH)) {
```

Reply:
```suggestion
    if (await this.configCore.hasFeature(FeatureFlag.DUPLICATE_DETECTION)) {
```
I'm going to also make some reviews adding the feature flag if I can do that here.

Reply:
@NicholasFlamy your change is wrong here.

Reply:
The condition needs to be inverted, right?

Reply:
Crap, I see what I did. Yes I think so.

Reply:
Oh yeah sorry that is right. Fixed :)

Reply:
Huh?

Reply:
I'll test it on my test branch.

Reply:
I don't know if it's inverted but regardless it's wrong. It removes the constant `machineLearning`.

Reply:
Yes that was the idea, as it isn't used afaict.

Reply:
> Yes that was the idea, as it isn't used afaict.

It's used below for the `maxDistance`.

Reply:
I think I am just blind lmao. Could you link the line for me please? :)